src/caffe/solver.cpp

```diff
@@ -775,9 +776,192 @@ void AdaGradSolver<Dtype>::ComputeUpdateValue(int param_id, Dtype rate) {
   }
 }
 
+template <typename Dtype>
+void AdaDeltaSolver<Dtype>::PreSolve() {
+  // Add the extra history entries for AdaDelta after those from
+  // SGDSolver::PreSolve
+  const vector<shared_ptr<Blob<Dtype> > >& net_params = this->net_->params();
```

Comment:
Are you trying to call `AdaDeltaSolver<Dtype>::PreSolve()` after calling `SGDSolver<Dtype>::Presolve()`?